eradiate/scenes/spectra.py

```diff
@@ -84,202 +102,117 @@ def f(value):
-        )
-
-
-@attr.s
-class RadianceMixin:
-    """Radiance quantity mixin."""
-    _quantity = "radiance"
-    _units_compatible = ureg.Unit("W/m^2/sr/nm")
-
-    @staticmethod
-    def _value_converter(val):
-        return ensure_units(val, cdu.generator("radiance"))
-
-
-@attr.s
-class IrradianceMixin:
-    """Irradiance quantity mixin."""
-    _quantity = "irradiance"
-    _units_compatible = ureg.Unit("W/m^2/nm")
-
-    @staticmethod
-    def _value_converter(val):
-        return ensure_units(val, cdu.generator("irradiance"))
+    """Uniform spectrum (*i.e.* constant against wavelength). Supports basic
+    arithmetics.
 
+    .. rubric:: Constructor arguments / instance attributes
 
-@attr.s
-class CollisionCoefficientMixin:
-    """Collision coefficient quantity mixin."""
-    _quantity = "collision_coefficient"
-    _units_compatible = ureg.Unit("m^-1")
-
-    @staticmethod
-    def _value_converter(val):
-        return ensure_units(val, cdu.generator("collision_coefficient"))
-
-
-@attr.s
-class AlbedoMixin:
-    """Albedo quantity mixin."""
-    _quantity = "albedo"
-    _units_compatible = ureg.Unit("dimensionless")
-
-    @staticmethod
-    def _value_converter(val):
-        return ensure_units(val, cdu.generator("albedo"))
-
+    Parameter ``value`` (float or :class:`~pint.Quantity`):
+        Uniform spectrum value. If a float is passed and ``quantity`` is not
+        ``None``, it is automatically converted to appropriate configuration
+        default units. If a :class:`~pint.Quantity` is passed and ``quantity``
+        is not ``None``, units will be checked for consistency.
 
-@attr.s
-class ReflectanceMixin:
-    """Reflectance quantity mixin."""
-    _quantity = "reflectance"
-    _units_compatible = ureg.dimensionless
+    See :class:`Spectrum` for undocumented members.
+    """
+    value = attrib_quantity(default=1.0)
+
+    @value.validator
+    def value_validator(self, attribute, value):
+        if self.quantity is not None and isinstance(value, pint.Quantity):
+            expected_units = cdu.get(self.quantity)
+
+            if not compatible(expected_units, value.units):
+                raise UnitsError(
+                    f"while validating {attribute.name}, got units "
+                    f"'{value.units}' incompatible with quantity {self.quantity} "
+                    f"(expected '{expected_units}')"
+                )
+
+        validator_is_positive(self, attribute, value)
+
+    def __attrs_post_init__(self):
+        if self.quantity is not None and self.value is not None:
+            self.value = ensure_units(self.value, cdu.get(self.quantity))
+
+    @property
+    def _values(self):
+        return self.value
+
+    def __add__(self, other):
+        # Preserve quantity field only if it is the same for both
+        # operands
+        if self.quantity is other.quantity and self.quantity is not None:
+            quantity = self.quantity
+        else:
```

Comment:
Why does this case have to be handled separately?

Reply:
I'm assuming you're referring to `self.quantity is not None`. Indeed, this can be removed.